pkg/kev/services.go

```diff
@@ -135,24 +135,25 @@ func (sc ServiceConfig) validate() error {
 	}
 
 	if !result.Valid() {
-		// Prioritise clear error messages.
+		// If we don't find anything useful just go with whatever is available.
+		err := errors.New(result.Errors()[0].Description())
+
 		for _, e := range result.Errors() {
+			// Prioritise clear error messages.
 			if e.Type() == "required" {
 				return errors.New(e.Description())
 			}
-		}
 
-		for _, e := range result.Errors() {
 			// These errors are very cryptic and hurt usability.
 			if e.Type() == "number_one_of" || e.Type() == "number_any_of" || e.Type() == "number_all_of" {
 				continue
 			}
 
-			return errors.New(e.Description())
+			// this is a valid error if there isn't anything better
+			err = errors.New(e.Description())
 		}
 
-		// If we don't find anything useful just go with whatever is available.
-		return errors.New(result.Errors()[0].Description())
+		return err
```

Comment:
and if you return above there is no need to return here and initialise err at the top of this funcion.

Reply:
responded in the previous comment